packages/synapse-react-client/src/components/TanStackTable/types.ts

```diff
@@ -0,0 +1,136 @@
+import { Cell, Row } from '@tanstack/react-table'
+import { RequireAllOrNone } from 'type-fest'
+import { TableBodyProps } from './TableBody'
+
```

Comment:
Making the slots work requires a fair amount of complex TypeScript. To keep the component files simple, I moved the types here. 

I don't think these are so complex that they are unmaintainable, but I do want to know if you think this is overengineered and if there's a better approach to take.

Reply:
Thank you for pulling out these types into a separate file.  These make sense, but could be a cognitive burden if defined in the component files